packages/shared/components/inputs/AliasInput.svelte

```diff
@@ -22,24 +23,24 @@
     $: alias = selected?.value
 
     export async function validate(): Promise<void> {
-        if (!alias) {
-            error = localize('error.aliasMinting.aliasRequired')
+        try {
+            if (!alias) {
+                throw new Error(localize('error.aliasMinting.aliasRequired'))
+            }
+            validateBech32Address($networkHrp, alias, ADDRESS_TYPE_ALIAS)
+            validateAliasInPossession()
+        } catch (err) {
+            error = err?.message ?? err
             return Promise.reject(error)
         }
-        const addressValidationError = validateBech32Address($networkHrp, alias, ADDRESS_TYPE_ALIAS)
-        if (addressValidationError) {
-            error = addressValidationError
-            return Promise.reject(error)
-        } else if (!isAliasInPossession()) {
-            error = localize('error.aliasMinting.aliasNotInPossession')
-            return Promise.reject(error)
-        } else {
-            return Promise.resolve()
-        }
     }
 
-    function isAliasInPossession(): boolean {
-        return aliasOptions.some((option) => option.value === alias)
+    function validateAliasInPossession(): void {
+        if (aliasOptions.some((option) => option.value === alias)) {
+            return
+        } else {
+            throw new Error(localize('error.aliasMinting.aliasNotInPossession'))
+        }
```

Comment:
```suggestion
    function validateAliasInPossession(): void {
        if (!aliasOptions.some((option) => option.value === alias)) {
            throw new Error(localize('error.aliasMinting.aliasNotInPossession'))
        }
```

we could even put this one right into `validate`